climada/hazard/tc_tracks.py

```diff
@@ -1293,6 +1292,9 @@ def _one_interp_data(track, time_step_h, land_geom=None):
             track_int.attrs['category'] = set_category(
                 track_int.max_sustained_wind.values,
                 track_int.max_sustained_wind_unit)
+            # restrict to time steps within original bounds
+            track_int = track_int.sel(time=((track.time[0] <= track_int.time)
+                                            & (track_int.time <= track.time[-1])))
```

Comment:
I'd find it more easy to read like that:
```suggestion
            track_int = track_int.sel(time=((track_int.time >= track.time[0])
                                            & (track_int.time <= track.time[-1])))
```
because you check whether `track_int.time` is within some bounds, but of course this has absolutely no impact on the result so you can keep as is, up to you.

Reply:
In Python, there is the shorthand notation `lower <= x <= upper` for scalar variables `x`. Unfortunately, there is no chaining of array comparisons in Python so far (see https://www.python.org/dev/peps/pep-0535/). That's why I'm always using the analogous ordering of expressions `(lower <= x_arr) & (x_arr <=upper)` for array variables `x_arr`.

I hope it's clearer when written in a single line of code:
https://github.com/CLIMADA-project/climada_python/blob/a33c91c54b867a5405798c06269a05033ed07a07/climada/hazard/tc_tracks.py#L1295-L1297